internal/worker.go

```diff
@@ -304,3 +292,24 @@ func ReplayWorkflowHistory(logger *zap.Logger, history *shared.History) error {
 	_, _, err := taskHandler.ProcessWorkflowTask(task, iterator)
 	return err
 }
+
+func extractHistoryFromFile(jsonfileName string) (*shared.History, error) {
+
+	raw, err := ioutil.ReadFile(jsonfileName)
+	if err != nil {
+		return nil, err
+	}
+
+	var deserializedEvents []shared.HistoryEvent
+	json.Unmarshal(raw, &deserializedEvents)
+
+	eventCount := len(deserializedEvents)
+	historyEvents := make([]*shared.HistoryEvent, eventCount, eventCount)
```

Comment:
Why is copy of history events needed here?

Reply:
Resolved